packages/vm/src/evm/interpreter.ts

```diff
@@ -87,9 +93,29 @@ export default class Interpreter {
   }
 
   async run(code: Buffer, opts: InterpreterOpts = {}): Promise<InterpreterResult> {
-    this._runState.code = code
-    this._runState.programCounter = opts.pc ?? this._runState.programCounter
+    if (
+      this._vm._common.isActivatedEIP(3540) &&
+      code.slice(0, 3).equals(Buffer.from('ef0001', 'hex'))
+    ) {
+      // Code is EOF1 format
+      const codeSections = eof1CodeAnalysis(code)
+      if (!codeSections) {
+        return {
+          runState: this._runState,
+          exceptionError: new VmError(ERROR.INVALID_EOF_FORMAT),
```

Comment:
Is this in the right place? These checks should be done at contract creation, right? Because this would OOG the `EF` contracts on mainnet (the 3 stray contracts)

Reply:
I just realized that the EIP also states that you can have EOF in `initCode`